en/sensor/rangefinders.md

```diff
@@ -0,0 +1,96 @@
+* [SF/LW20](http://lightware.co.za/shop2017/drone-altimeters/51-lw20-100-m.html) (100 m) - Waterproofed (IP67) with servo for sense-and-avoid applications
+
+Drivers exist for both I2C and serial ports, which can be configured using the parameters [SENS_EN_SF0X](../advanced_config/parameter_reference.md#SENS_EN_SF0X) and [SENS_EN_SF1XX](../advanced_config/parameter_reference.md#SENS_EN_SF1XX) (respectively).
+
+Wiring and other information for the I2C variants can be found in the topic [Lightware SFxx Lidar](../sensor/sfxx_lidar.md).
+
+> **Tip** Not all devices are supported for both serial and I2C; check the parameters for more information and supported models.
+
+
+### TeraRanger One
+
+TeraRanger provide a number of lightweight distance measurement sensor based on infrared Time-of-Flight (ToF) technology. They are typically faster and have greater range than sonar, and smaller and lighter than laser-based systems. PX4 supports:
+* [TeraRanger One](http://www.teraranger.com/products/teraranger-one/) (0.2 - 14 m)
+* [TeraRanger Evo](https://www.terabee.com/portfolio-item/teraranger-evo-infrared-distance-sensor/) (0.5 – 60 m)
+
+These rangefinders must be connected via the I2C bus (using an [I2C adapter](http://www.teraranger.com/product/teraranger-i2c-adapter/)).
+
+The sensors are enabled using the parameter [SENS_EN_TRANGER](../advanced_config/parameter_reference.md#SENS_EN_TRANGER) (you can set the type of sensor or that PX4 should auto-detect the type).
+
+
+### uLanding Radar
+
+The *Aerotenna* [uLanding](https://aerotenna.com/sensors/#ulanding) altimeter is compact microwave rangefinder that has been optimised for use on UAVs. It has a sensing range of 45m. A particular advantages of this product are that it can operate effectively in all weather conditions and over all terrain types (including water).
+
+The uLanding Radar is not present in "most" firmware by default and must be started by updating a configuration file (rather than by a parameter). More information can be found here: [uLanding Radar](../sensor/ulanding_radar.md).
+
+
+### Other
+
+PX4 also supports the Bebop rangefinder.
+
+
+## Rangefinder Configuration
+
+The rangefinder is configured using [EKF2\_RNG\_*](../advanced_config/parameter_reference.md#EKF2_RNG_AID) parameters, e.g. delay, offset of rangefinder from vehicle body, etc.
+
+
+## Testing
+
+Rangefinder values/performance can be tested using the *QGroundControl Analyse tool*, or in the *QGroundControl MAVLink Console* by observing the `distance_sensor` uORB topic:
+```sh
+listener distance_sensor 5
+```
+
+For more information see: [Sensor/Topic Debugging using the Listener Command](https://dev.px4.io/en/debug/sensor_uorb_topic_debugging.html) (PX4 Development Guide).
+
+
+## Simulation
+
+Lidar and sonar rangefinders can be used in the [Gazebo Simulator](https://dev.px4.io/en/simulation/gazebo.html) (PX4 Development Guide).
+
+<!-- 

```

Comment:
@ChristophTobler How do you enable lidar/sonar in Gazebo?

Reply:
You have to use the iris model that includes the rangefinder -> `make posix_sitl_default gazebo_iris_opt_flow` (or include the sensor in another model, see https://github.com/PX4/sitl_gazebo/blob/master/models/iris_opt_flow/iris_opt_flow.sdf#L8-L22). If you don't want to use the rangefinder with GPS change `EKF2_AID_MASK` to `1`.

Reply:
Thanks very much. I will update appropriately!